lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2636,6 +2640,22 @@ void AttributeChecker::visitCompilerEvaluableAttr(CompilerEvaluableAttr *attr) {
   // TypeChecker::checkFunctionBodyCompilerEvaluable().
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+void AttributeChecker::visitTensorFlowGraphAttr(TensorFlowGraphAttr *attr) {
+  const FuncDecl *FD = cast<FuncDecl>(D);
+  // The function must be top-level.
+  if (FD->getImplicitSelfDecl()) {
+    diagnoseAndRemoveAttr(attr, diag::tf_graph_attr_top_level_only);
+    return;
+  }
+  // Only functions taking and returning TensorFlow values are permitted.
+  if (!tf::isTensorFlowValueOrAggregate(
+        FD->getParameterList(0)->getType(FD->getASTContext())) ||
+      !tf::isTensorFlowValueOrAggregate(FD->getResultInterfaceType()))
+    diagnoseAndRemoveAttr(attr,
+                          diag::tf_graph_attr_function_tensorflow_value_only);
+}
```

Comment:
Is it reasonable to set @convention(tensorflow) on the function's type here?

Reply:
There's no setter for a function's `ExtInfo` unfortunately. I think it's better to follow how Swift assigns `@convention(thin)` to normal top-level functions, which is what I did in `lib/SIL/SILFunctionType.cpp`.